src/ophyd_async/core/signal.py

```diff
@@ -45,17 +45,20 @@ class Signal(Device, Generic[T]):
     """A Device with the concept of a value, with R, RW, W and X flavours"""
 
     def __init__(
-        self, backend: SignalBackend[T], timeout: Optional[float] = DEFAULT_TIMEOUT
+        self,
+        backend: SignalBackend[T],
+        timeout: Optional[float] = DEFAULT_TIMEOUT,
+        name: str = "",
     ) -> None:
-        self._name = ""
+        self._name = name
         self._timeout = timeout
         self._init_backend = self._backend = backend
 
     @property
     def name(self) -> str:
         return self._name
 
-    def set_name(self, name: str = ""):
+    def set_name(self, name: str):
         self._name = name
```

Comment:
Device already has `set_name` and `name`, so you can delete these and do `super().__init__(name)`

Reply:
Okay, have made that change